tools/cpu_graph_visualizer/visualizer.py

```diff
@@ -0,0 +1,586 @@
+
+def fill_tensors_with_random(input, shape):
+    dtype = get_dtype(input.get_element_type())
+    rand_min, rand_max = (0, 1) if dtype == np.bool else (np.iinfo(np.uint8).min, np.iinfo(np.uint8).max)
+    # np.random.uniform excludes high: add 1 to have it generated
+    if np.dtype(dtype).kind in ['i', 'u', 'b']:
+        rand_max += 1
+    rs = np.random.RandomState(np.random.MT19937(np.random.SeedSequence(0)))
+    a = rs.uniform(rand_min, rand_max, list(shape)).astype(dtype)
+    return ov.Tensor(a)
+
+def test_infer_queue(compiled_model, input_shapes, num_request, num_infer, time_limit=60):
+    infer_queue = ov.AsyncInferQueue(compiled_model, num_request)
+
+    latency_list = []
+    prof_list = []
+    def callback(request, userdata):
+        latency_list.append(request.latency)
+        prof_list.append(request.profiling_info)
+
+    infer_queue.set_callback(callback)
+
+    all_input = {}
+    for port, input in enumerate(compiled_model.inputs):
+        if port < len(input_shapes):
+            static_shape = input_shapes[port]
+        else:
+            static_shape = input.get_shape()
+        all_input[port] = fill_tensors_with_random(input, static_shape)
+
+    for i in range(num_request):
+        infer_queue.start_async(all_input, userdata=i)
+
+    t0 = time.time()
+    for i in range(num_infer):
+        wtime = time.time() - t0
+        if time_limit and (wtime > time_limit):
+            break
+        infer_queue.start_async(None, userdata=i)
+    infer_queue.wait_all()
+    fps = i/wtime
+    return latency_list, prof_list, fps, wtime
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser("CPU graph visualizer")
+    parser.add_argument("model", type=str, help="Model file path")
+    parser.add_argument("-p","--perf", action="store_true", help="do profiling")
+    parser.add_argument("--raw", action="store_true", help="also dump raw ngraph model")
+    parser.add_argument("--bf16", action="store_true", help="enable inference precision with bf16")
+    parser.add_argument("--nthreads", type=int, default=4, help="set INFERENCE_NUM_THREADS for profiling")
+    parser.add_argument("-r","--reshape", type=str, default="()", help="reshape before visualize/profiling")
+    parser.add_argument("-i","--input_shapes", type=str, default="()",
+                        help="python tuple/list of static shapes used for profiling, e.g. [[6,23,240],[5,10,2]]")
+    args = parser.parse_args()
+
+    core = ov.Core()
+    model_path = args.model
+    model_fname = os.path.splitext(os.path.split(model_path)[1])[0]
+    model = core.read_model(model_path)
+
+    device = "CPU"
+
+    dev_prop = {"PERF_COUNT": "YES",
+                "AFFINITY": "CORE",
+                "PERFORMANCE_HINT_NUM_REQUESTS":0,
+                "PERFORMANCE_HINT":"",
+                "INFERENCE_PRECISION_HINT": "bf16" if args.bf16 else "f32",
+                "NUM_STREAMS" : 1,
+                "INFERENCE_NUM_THREADS" : args.nthreads}
```

Comment:
```suggestion
    dev_prop = {"PERF_COUNT": "YES",
                "AFFINITY": "CORE",
                "PERFORMANCE_HINT_NUM_REQUESTS":1,
                "PERFORMANCE_HINT":"UNDEFINED",
                "INFERENCE_PRECISION_HINT": "bf16" if args.bf16 else "f32",
                "NUM_STREAMS" : 1,
                "INFERENCE_NUM_THREADS" : args.nthreads}
```
Or you can remove `PERFORMANCE_HINT` as there is no setting here anyway.